### DISCUSSION TITLE
Where do I get the required credentials?

### DISCUSSION THREAD

**Discussion by saleweaver** [saleweaver/python-amazon-sp-api#35] [FAQ] [answered]
I'm struggling on where to obtain all the credential keys:

refresh_token: ''
lwa_app_id: ''
lwa_client_secret: ''
aws_secret_key: ''
aws_access_key: ''
role_arn: ''

Can you provide documentation/guidance on where these are obtained from.

**Comment by saleweaver** [comment, marked as answer]
This is best explained in Amazon's documentation, found here:

https://github.com/amzn/selling-partner-api-docs/blob/main/guides/developer-guide/SellingPartnerApiDeveloperGuide.md#registering-your-selling-partner-api-application

**Reply by harvetech** [reply]
@saleweaver  this link so no longer working. Can you please update it ? I got all credentials but was unable to find the refresh tokken. Can you please explain it a bit? 
Thank you very much.

**Reply by tector** [reply]
I found out that one way you can create the refresh token is inside the selling partner developer console (https://sellercentral-europe.amazon.com/sellingpartner/developerconsole) where you have created an app.

This is the only way i understand (for now) and it is documented here:
https://github.com/amzn/selling-partner-api-docs/blob/main/guides/en-US/developer-guide/SellingPartnerApiDeveloperGuide.md#self-authorization
